ape_alchemy/provider.py

```diff
@@ -5,22 +5,19 @@
 from ape.exceptions import ContractLogicError, ProviderError, VirtualMachineError
 from evm_trace import CallTreeNode, ParityTraceList, TraceFrame, get_calltree_from_parity_trace
 from requests import HTTPError
-from web3 import HTTPProvider, Web3  # type: ignore
+from web3 import HTTPProvider, Web3
 from web3.exceptions import ContractLogicError as Web3ContractLogicError
 from web3.gas_strategies.rpc import rpc_gas_price_strategy
 from web3.middleware import geth_poa_middleware
 
 from .exceptions import AlchemyFeatureNotAvailable, AlchemyProviderError, MissingProjectKeyError
 
-_ENVIRONMENT_VARIABLE_OPTIONS = {
-    "ethereum": ("WEB3_ALCHEMY_PROJECT_ID", "WEB3_ALCHEMY_API_KEY"),
-    "arbitrum": ("WEB3_ARBITRUM_ALCHEMY_PROJECT_ID", "WEB3_ARBITRUM_ALCHEMY_API_KEY"),
-    "optimism": ("WEB3_OPTIMISM_ALCHEMY_PROJECT_ID", "WEB3_OPTIMISM_ALCHEMY_API_KEY"),
-    "polygon": ("WEB3_POLYGON_ALCHEMY_PROJECT_ID", "WEB3_POLYGON_ALCHEMY_API_KEY"),
-}
+# The user must either set one of these or an ENV VAR of the pattern:
+#  WEB3_<ECOSYSTEM>_<NETWORK>_PROJECT_ID or  WEB3_<ECOSYSTEM>_<NETWORK>_API_KEY
+DEFAULT_ENVIRONMENT_VARIABLE_NAMES = ("WEB3_ALCHEMY_PROJECT_ID", "WEB3_ALCHEMY_API_KEY")
 
 
-class AlchemyEthereumProvider(Web3Provider, UpstreamProvider):
+class Alchemy(Web3Provider, UpstreamProvider):
```

Comment:
Alchemy also has starknet too right?

Reply:
It does! But I am not sure how well it works.

Reply:
Also I don't see it in the drop down for options in the Alchemy web app, so maybe it is not really supported yet?